src/main/java/com/example/demo/services/BookingService.java

```diff
@@ -0,0 +1,78 @@
+package com.example.demo.services;
+
+import com.example.demo.entities.BookingEntity;
+import com.example.demo.entities.CustomerEntity;
+import com.example.demo.entities.LocationEntity;
+import com.example.demo.exceptions.BookingNotFoundException;
+import com.example.demo.exceptions.LocationNotFoundException;
+import com.example.demo.repositories.BookingEntityRepository;
+import com.example.demo.repositories.CustomerEntityRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class BookingService {
+
+    @Autowired private BookingEntityRepository bookingEntityRepository;
+    @Autowired private CustomerEntityRepository customerEntityRepository;
+
+    public BookingEntity createBooking(BookingEntity booking) {
+        return bookingEntityRepository.save(booking);
+    }
+
+    public BookingEntity findBookingById(Long id) {
+
+        if(!bookingEntityRepository.existsById(id)) {
+            { throw new BookingNotFoundException(); }
+        }
+        return bookingEntityRepository.findById(id).get();
```

Comment:
If you want to get an entity and throw an exception if it didn't;t exist, rather than perform two repo operations you can do it in one with findById(id).orElseThrow(Exception:new)